sdk/scheduler/src/main/java/com/mesosphere/sdk/framework/OfferProcessor.java

```diff
@@ -217,13 +226,42 @@ private void processQueuedOffers() {
         }
     }
 
-    private void evaluateOffers(List<Protos.Offer> offers) {
-        if (isDeregistered.get()) {
-            // Just in case there are unflushed offers in the queue when we're deregistered
-            LOGGER.info("Dropping {} queued offers: Framework is deregistered", offers.size());
-            return;
+    private boolean checkStatuses() {
+        StatusResponse statusResponse = mesosEventClient.status();
```

Comment:
So, in the MSScheduler case, this is the global status across all services, right?

Reply:
Yes, this is returned by `MultiServiceEventClient`, which in practice would only return `uninstalled` or `running`, with `uninstalled` being "all services have been wound down, now the scheduler can be taken down"